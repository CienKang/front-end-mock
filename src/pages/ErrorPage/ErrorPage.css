.error-container {
    font-size: 4rem;
    height: 90vh;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-left: 20px;
    gap: 30px;
}

.error-container button {
    width: 150px;
    height: 50px;
    border-radius: 10px;
    border: none;
    background-color: #4c63aa;
    color: white;
    font-size: 1.5rem;
}