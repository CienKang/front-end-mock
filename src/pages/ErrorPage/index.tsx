import React from 'react';
import { useNavigate } from 'react-router-dom';
import './ErrorPage.css';
const ErrorPage = () => {

    const navigate = useNavigate();
    const handleSyncClick = () => {
        navigate('/');
    };

    return (
        <div className='error-container'>
            <p>:((</p>
            <p>seems a bit empty in here...</p>
            <button onClick={handleSyncClick}>sync</button>
        </div>
    );
};

export default ErrorPage;