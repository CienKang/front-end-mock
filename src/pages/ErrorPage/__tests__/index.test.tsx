import React  from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import ErrorPage from '../index';
import * as router from 'react-router';

const navigate = jest.fn();
beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
});

describe('ErrorPage', () => {

    it('should render the component correctly with error message shown and a button when encountered any error',()=>{
        render(<ErrorPage/>);
        expect(screen.getByText(/:\(\(/i)).toBeTruthy();
        expect(screen.getByText(/seems a bit empty in here.../i)).toBeTruthy();
        expect(screen.getByRole('button', {  name: /sync/i})).toBeTruthy();
    });

    it('should redirect to main page when the sync button is clicked',()=>{
        render(<ErrorPage />);
        expect(navigate).not.toBeCalled();
        fireEvent.click(screen.getByRole('button', {  name: /sync/i}));
        expect(navigate).toBeCalledWith('/');
    });

    it('should match with the previous snapshots',()=>{
        const {asFragment} = render(<ErrorPage/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
