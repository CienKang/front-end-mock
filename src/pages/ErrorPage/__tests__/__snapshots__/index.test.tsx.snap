// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should match with the previous snapshots 1`] = `
<DocumentFragment>
  <div
    class="error-container"
  >
    <p>
      :((
    </p>
    <p>
      seems a bit empty in here...
    </p>
    <button>
      sync
    </button>
  </div>
</DocumentFragment>
`;
